File: ub/pset6/readability/readability.py
from cs50 import get_string

letters = 0
words = 0
sentences = 0

def main():
    global letters
    global words
    global sentences
    #prompt user for information
    text = get_string("Text:" )
    
    
    for i in range(len(text)):
        
        if i > len(text):
            break
        #check if spaces
        if not " " in text[i]:
            #check if not punction
            if not text[i] in ('!', "," ,"\'" ,";" ,"\"", ".", "-" ,"?"):
                letters += 1
            else:
                if text[i] == "." or text[i] == "?" or text[i] == "!":
                    sentences += 1
        #if is space
        word = text.split()
        words = len(word)
    ColemanLiau(AverageLtoW(),AverageStoW())
    





def ColemanLiau(l , s):
    
    letterCount = 0.0588 * l
    wordCount = 0.296 * s
    coleman = (letterCount - wordCount - 15.8)
    #check if under level 1
    index = round(coleman)
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Grade " + str(index))

def AverageLtoW():
    average = 0
    l = float(letters)
    w = float(words)
    #get average letter to word
    average = (l/w) * 100
    return average

def AverageStoW():
    average = 0
    
    s = float(sentences)
    w = float(words)
    # get average letter to word
    average = (s/w)*100
    return average


main()

File: ub/pset1/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
void ColemanLiau(float l, float s);
float AverageLtoW();
float AverageStoW();

int letters = 0;
int words = 0;
int sentences = 0;
int main(void)
{
    //prompt user for text
    string text = get_string("Text: %c ", isprint(13));
    // loop through the text 
  
    for (int i = 0; i < strlen(text); i++)
    {
      
    
        //check if not a space
        if (!isspace(text[i]))
        {
            //if not punctation either then increase the letter
            if (!ispunct(text[i]))
            {
                letters++;
            }
            else
            {
                //if next letter
                if (text[i] == '.' || text[i] == '?' || text[i] == '!')
                {
                    sentences++;
                }
            }
        }
        //is a space
        else
        { 
            //check if space is followed by a letter
            if (isalpha(text[i + 1]) || !ispunct(text[i]))
            {
               
                words++;
          
            }
        
        }
        
        if (i == strlen(text) - 1)
        {
            words++;
        }
    }
    ColemanLiau(AverageLtoW(), AverageStoW());
 

}

void ColemanLiau(float l, float s)
{
    float letterCount = 0.0588 * l;
    float wordCount = 0.296 * s;
    float coleman = (letterCount - wordCount) - 15.8;
    int index = round(coleman);
    //check if level under 1
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    //check if index is higher then 16
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
   
}
 
float AverageLtoW()
{
    float average;
    //cast int to float
    float l = (float)letters;
    float w = (float)words;
    //get average letter to word
    average = (l / w) * 100;
     
    return average;
    
}

float AverageStoW()
{
    float average;
    //cast int to float
    float s = (float)sentences;
    float w = (float)words;
    //get average letter to word
    average = (s / w) * 100;
    return average;
}

